fair_mango/metrics/metrics.py

```diff
@@ -697,3 +700,280 @@ def is_biased(self, threshold: float = 0.1) -> dict[str, bool]:
                 bias[target] = False
 
         return bias
+
+
+def super_set_fairness_metrics(
+    metric: (
+        type[DemographicParityDifference]
+        | type[DemographicParityRatio]
+        | type[DisparateImpactDifference]
+        | type[DisparateImpactRatio]
+        | type[EqualOpportunityDifference]
+        | type[EqualOpportunityRatio]
+        | type[EqualisedOddsDifference]
+        | type[EqualisedOddsRatio]
+        | type[FalsePositiveRateDifference]
+        | type[FalsePositiveRateRatio]
+    ),
+    data: Dataset | pd.DataFrame,
+    sensitive: Sequence[str] = [],
+    real_target: Sequence[str] | None = None,
+    predicted_target: Sequence[str] | None = None,
+    positive_target: Sequence[int | float | str | bool] | None = None,
+) -> list:
+    """Calculate fairness metrics for different subsets of sensitive
+    attributes. Ex:
+    [gender, race] → (gender), (race), (gender, race)
+
+    Parameters
+    ----------
+    metric : type[DemographicParityDifference]  |  type[DemographicParityRatio]
+    |  type[DisparateImpactDifference]  |  type[DisparateImpactRatio]  |
+    type[EqualOpportunityDifference]  |  type[EqualOpportunityRatio]  |
+    type[EqualisedOddsDifference]  |  type[EqualisedOddsRatio]  |
+    type[FalsePositiveRateDifference]  |  type[FalsePositiveRateRatio]
```

Comment:
why so much space between type hints ?

Reply:
I believe it is just how GitHub formats the text. In VSCode it appears like this
![Screenshot 2024-07-16 at 14 46 33](https://github.com/user-attachments/assets/0df3234d-41e1-47d8-810d-43b362dd64da)